### PR TITLE
Add batches executor to dev env

### PR DESCRIPTION
**Pull request by mrnugget**
Part of https://github.com/sourcegraph/sourcegraph/issues/22550

Based on: https://github.com/sourcegraph/sourcegraph/pull/22555

### CODE REVIEW

File: dev/Procfile [hunk 1]
```diff
@@ -25,6 +25,7 @@ postgres_exporter: ./dev/postgres_exporter.sh
 # Enterprise executor services
 executor-queue: test "$ENTERPRISE" != 1 || executor-queue
 codeintel-executor: test "$ENTERPRISE" != 1 || env EXECUTOR_QUEUE_NAME=codeintel TMPDIR=$HOME/.sourcegraph/indexer-temp executor
+batches-executor: test "$ENTERPRISE" != 1 || env EXECUTOR_QUEUE_NAME=batches TMPDIR=$HOME/.sourcegraph/batches-executor-temp executor
```
Comment (eseliger) [inline dev/Procfile:28]: One unfortunate compromise of not naming these like the cmd they run (`executor`) is that they don't get auto restart. It works in sg though, so not a blocker I think 😬

File: sg.config.yaml [hunk 2]
```diff
@@ -382,20 +382,32 @@ commands:
...
       - lib
       - internal
       - enterprise/internal
       - enterprise/cmd/executor
 
+  codeintel-executor:
+    <<: *executor_template
+    env:
+      EXECUTOR_QUEUE_NAME: codeintel
+      TMPDIR: $HOME/.sourcegraph/indexer-temp
```
Comment (eseliger) [inline sg.config.yaml:403]: neat neat neat

**Review by eseliger — APPROVED** [review]

File: sg.config.yaml (hunk 2 above)
Comment (mrnugget) [reply sg.config.yaml]: Had to change this a bit, since the `TMPDIR` if set for the `install` process kinda breaks `go build`.

**Comment by ryanslade** [comment]
This seems to break our local dev environment when not using `sg`.

`./enterprise/dev/start.sh` fails because both the code intel and batches executors are trying to bind to the same port, 6092

**Comment by eseliger** [comment]
Looking!

**Pull request merged**